File: Java-Core/src/main/java/presto/javascript/JavaScriptNativeCategoryBinding.java
package presto.javascript;

import presto.grammar.NativeCategoryBinding;
import presto.utils.CodeWriter;


public class JavaScriptNativeCategoryBinding extends NativeCategoryBinding {

	String identifier;
	JavaScriptModule module;
	
	public JavaScriptNativeCategoryBinding(String identifier, JavaScriptModule module) {
		this.identifier = identifier;
		this.module = module;
	}

	@Override
	public void toDialect(CodeWriter writer) {
		writer.append("JavaScript: ");
		writer.append(identifier);
		if(module!=null)
			module.toDialect(writer);
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/javascript/JavaScriptNativeCategoryBinding.js
var NativeCategoryBinding = require("./../grammar/NativeCategoryBinding").NativeCategoryBinding;

function JavaScriptNativeCategoryBinding(identifier, module) {
	NativeCategoryBinding.call(this);
	this.identifier = identifier;
	this.module = module || null;
	return this;
}

JavaScriptNativeCategoryBinding.prototype = Object.create(NativeCategoryBinding.prototype);
JavaScriptNativeCategoryBinding.prototype.creator = JavaScriptNativeCategoryBinding;

JavaScriptNativeCategoryBinding.prototype.resolve = function() {
	var m = this.resolve_module();
	if(m==null) {
		return eval(this.identifier);
	} else {
		return m[this.identifier] || null;
	}
};

JavaScriptNativeCategoryBinding.prototype.resolve_module = function(context) {
	if (this.module == null) {
		return null;
	} else {
		return this.module.resolve();
	}
};

JavaScriptNativeCategoryBinding.prototype.toDialect = function(writer) {
    writer.append("JavaScript: ");
    writer.append(this.identifier);
    if(this.module!=null)
        this.module.toDialect(writer);
};

exports.JavaScriptNativeCategoryBinding = JavaScriptNativeCategoryBinding;

